### PR TITLE
add support for datadog logs

### PR DESCRIPTION
**Pull request by ludov04**
Please be critical, as it is my first PR and experience with golang

### CODE REVIEW

File: provider/ecs/ecs.go [hunk 1]
```diff
@@ -319,6 +321,15 @@ func (e *ECS) CreateTaskDefinition(d service.Deploy) (*string, error) {
 					"awslogs-stream-prefix": aws.String(container.ContainerName),
 				},
 			})
+		} else if util.GetEnv("DATADOG_LOGS_ENABLED", "no") == "yes" {
+			containerDefinition.SetLogConfiguration(&ecs.LogConfiguration{
+				LogDriver: aws.String("json-file"),
+				Options:   map[string]*string{},
```
Comment (pollett) [inline provider/ecs/ecs.go:327]: You can probably drop this line completely

File: provider/ecs/ecs.go [hunk 2]
```diff
@@ -319,6 +321,15 @@ func (e *ECS) CreateTaskDefinition(d service.Deploy) (*string, error) {
...
+			})
+			if containerDefinition.DockerLabels["com.datadoghq.ad.logs"] == nil {
+				datadogLogConfig := fmt.Sprintf(`[{ "source": "%s", "service": "%s" }]`, "docker", e.ServiceName)
```
Comment (pollett) [inline provider/ecs/ecs.go:330]: Given this is simple json I think this is fine, but if it was more complicated you might consider using a struct and converting it to json, along the lines of

```
  jstr, _ := json.Marshal(struct {
    Source  string  `json:"source"`
    Service string  `json:"service"`
  }{
    "docker",
    e.ServiceName,
  })
```

**Review by pollett — COMMENTED** [review]
I think the changes look good, the greater question will be whether we're happy with ecs-deploy turning  into something DataCamp specific, i.e. with our particular datadog settings built in, or whether we should inject these into the config that we pass to ecs-deploy.

The other option would be to say in our docker container have a default.json that's merged with the ecs.json before being sent to ecs-deploy. 

I'm not sure on the answer to this though, any thoughts?

**Comment by ludov04** [comment]
This was done outside ecs-deploy

**Pull request closed without merging**